### PR TITLE
Support attribute access from other modules

### PR DESCRIPTION
**Pull request by farahhariri**
Accessing an attribute from another module currently fails if it has a complex data type like string. This PR passes the builder object to the `get_getattr` function and uses `get_constant_generic` instead of `get_constant` to retrieve attributes. A test case that shows an example that would usually fail and this PR fixes is added

### CODE REVIEW

**Comment by esc** [comment]
@farahhariri thanks for submitting this! Perhaps you could provide a little more context as to why this is needed or what it fixes? Thanks!

**Comment by farahhariri** [comment]
> @farahhariri thanks for submitting this! Perhaps you could provide a little more context as to why this is needed or what it fixes? Thanks!

Hi @esc This is needed for when someone accesses an attribute (like a constant tuple) in another module. I will add a test case to it to show what it addresses

**Comment by esc** [comment]
@farahhariri excellent, test cases are always welcome.

File: numba/tests/test_objects.py [hunk 1]
```diff
@@ -30,6 +29,11 @@ def delattr_usecase(o):
 
 
 class TestAttributes(TestCase):
+    def test_getattr(self, flags=enable_pyobj_flags):
+        import numba.tests.test_analysis as test_analysis
+        def f():
+            return test_analysis._GLOBAL
```
Comment (ehsantn) [inline numba/tests/test_objects.py:35]: I think the global needs to be a type like string that would need the builder for lowering.

**Review by ehsantn — APPROVED** [review]
Looks good to me. Thanks.

**Review by sklam — APPROVED** [review]
Thanks for the contribution.

**Comment by stuartarchibald** [comment]
@farahhariri Congratulations on your first contribution to Numba!

**Comment by farahhariri** [comment]
> @farahhariri Congratulations on your first contribution to Numba!

Thanks @stuartarchibald :)

**Pull request merged**